doc/OnlineDocs/contributed_packages/mindtpy.rst

```diff
@@ -71,67 +107,179 @@ MindtPy also supports setting options for mip solver and nlp solver.
+  ...                                    mip_solver='cplex',
+  ...                                    nlp_solver='ipopt',
+  ...                                    add_regularization='level_L1' 
+  ...                                    # alternative regularizations
+  ...                                    # 'level_L1', 'level_L2', 'level_L_infinity', 
+  ...                                    # 'grad_lag', 'hess_lag', 'hess_only_lag', 'sqp_lag'
+  ...                                    )
+  >>> model.objective.display()
 
-  >>> model.x = pyo.Var(bounds=(1.0, 10.0), initialize=5.0)
-  >>> model.y = pyo.Var(within=Binary)
 
-  >>> model.c1 = Constraint(expr=(model.x-4.0)**2 - model.x <= 50.0*(1-model.y))
-  >>> model.c2 = pyo.Constraint(expr=model.x*log(model.x)+5.0 <= 50.0*(model.y))
-  
-  >>> model.objective = pyo.Objective(expr=model.x, sense=pyo.minimize)
-
-  Solve the model using single tree implementation in MindtPy
-  >>> pyo.SolverFactory('mindtpy').solve(
-  ...    model, strategy='OA',
-  ...    mip_solver='cplex_persistent', nlp_solver='ipopt', single_tree=True)
+Solution pool implementation
+----------------------------
+
+MindtPy supports solution pool of the MILP solver, CPLEX and GUROBI. With the help of the solution, MindtPy can explore several integer combinations in one iteration. 
+
+A usage example for OA with solution pool is as follows:
+
+.. code::
+
+  >>> pyo.SolverFactory('mindtpy').solve(model,
+  ...                                    strategy='OA',
+  ...                                    mip_solver='cplex_peristent',
+  ...                                    nlp_solver='ipopt',
+  ...                                    solution_pool=True,
+  ...                                    num_solution_iteration=10 # default=5
+  ...                                    tee=True
+  ...                                    )
+  >>> model.objective.display()
+
+Feasibility Pump
+----------------
+
+For some MINLP problems, Outer Approximation method might have difficulty in finding a feasible solution. MindtPy provides the Feasibility Pump implementation to quickly find feasible solutions for convex MINLPs. The main idea of Feasibility Pump is to decompose the original mixed-integer problem into two parts: integer feasibility and constraint feasibility. For convex MINLPs, a MIP is solved to obtain a solution, which satisfies the integrality constraints on y, but may violate some of the nonlinear constraints; next, by solving an NLP, a solution is computed that satisfies the nonlinear constraints but might again violate the integrality constraints on y. By minimizing iteratively the distance between these two types of solutions, a solution that is both constraint and integer feasible can be expected. In MindtPy, feasibility pump can be used both as a initialization strategy and a decomposition strategy. For details of this implementation are included in [`Bernal et al., 2017`_].
```

Comment:
For some MINLP problems, the Outer Approximation method might have difficulty in finding a feasible solution. MindtPy provides the Feasibility Pump implementation to find feasible solutions for convex MINLPs quickly. The main idea of the Feasibility Pump is to decompose the original mixed-integer problem into two parts: integer feasibility and constraint feasibility. For convex MINLPs, a MIP is solved to obtain a solution, which satisfies the integrality constraints on y, but may violate some of the nonlinear constraints; next, by solving an NLP, a solution is computed that satisfies the nonlinear constraints but might again violate the integrality constraints on `y`. By minimizing the distance between these two types of solutions iteratively, a constraint and integer feasible solution can be expected. In MindtPy, the Feasibility Pump can be used both as an initialization strategy and a decomposition strategy. For details of this implementation are included in [`Bernal et al., 2017`_].